### PR TITLE
Revert me: patches to make build on Ubuntu Cosmic 18.10

### LINKED ISSUE #56: sumo branch build failure
Build Configuration:
BB_VERSION           = "1.38.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "i686-poky-linux"
MACHINE              = "edison"
DISTRO               = "poky-edison"
DISTRO_VERSION       = "2.5.2"
TUNE_FEATURES        = "m32 corei7"
TARGET_FPU           = ""
meta
meta-poky
meta-yocto-bsp       = "sumo:623b77885051174d0e05198843e739110977bd18"
meta-oe
meta-python
meta-networking      = "sumo:8760facba1bceb299b3613b8955621ddaa3d4c3f"
meta-intel           = "sumo:90af97d23fb2a56187c2fe2a3f4f4190d7cc2605"
meta-intel-edison-bsp
meta-intel-edison-distro = "master:d0de39d612c6cb032f19acd3b89f2f4264c444b9"
meta-acpi            = "eds:477a31d2bad06de9c240753462695c46569cf162"

ERROR: Multiple versions of openssl are due to be built (/home/cloud/edison/out/linux64/poky/meta/recipes-connectivity/openssl/openssl_1.1.0i.bb /home/cloud/edison/out/linux64/poky/meta/recipes-connectivity/openssl/openssl_1.0.2p.bb). Only one version of a given PN should be built in any given build. You likely need to set PREFERRED_VERSION_openssl to select the correct version or don't depend on multiple versions.

Summary: There was 1 ERROR message shown, returning a non-zero exit code.
Makefile:53: recipe for target 'edison-image' failed
make: *** [edison-image] Error 1

### PR DESCRIPTION
**Pull request by htot**
These patches are needed to build on Ubuntu Cosmic due to glibc 2.28. I tested this on Cosmic and on Xenial to verify backwards compatibility. It has been independantly tested by @sjpeng. I expect at some point these patches will be backported to Sumo at which point this will break and need to be reverted.

This closes #56

### CODE REVIEW

File: meta-intel-edison-distro/conf/distro/poky-edison.conf [hunk 1]
```diff
@@ -7,7 +7,6 @@ PREFERRED_PROVIDER_virtual/bootloader ?=  "u-boot"
 PREFERRED_VERSION_u-boot ?= "2018.09"
 PREFERRED_VERSION_u-boot-fw-utils ?= "2018.09"
 PREFERRED_VERSION_pulseaudio ?= "11.1%"
-PREFERRED_VERSION_openssl ?= "1.1.0%"
```
Comment (alext-mkrs) [inline meta-intel-edison-distro/conf/distro/poky-edison.conf]: I think this one needs a closer and a separate look - as this should have nothing to do with the glibc problems other patches in the PR solve. And downgrading the OpenSSL doesn't look good to me. Let me try to look into this while I'm prepraring a build for #50.

Comment (htot) [reply meta-intel-edison-distro/conf/distro/poky-edison.conf]: No, apparently something depends on openssl 1.0.2p. Which it didn't before.

Comment (alext-mkrs) [reply meta-intel-edison-distro/conf/distro/poky-edison.conf]: Yeah, I understood from your discussion in #56 that's something unexpected and possibly related to sumo updates, however it's a bit hard to believe they'd change something in favor of downgrading the OpenSSL, so I think that may be some bug/misconfig somewhere we could chase.

Comment (htot) [reply meta-intel-edison-distro/conf/distro/poky-edison.conf]: So what do think? 

BTW if we can pull this then we have a more or less stable sumo. The new thud branch that I will send a PR for next has openssl 1.1 really. And U-Boot 2019.01, kernel 4.20. Is much more interresting :-)

Comment (alext-mkrs) [reply meta-intel-edison-distro/conf/distro/poky-edison.conf]: Just got to investigate that - right, what you've outline is exactly the reason. Let's then just go with removing the preferred version indeed for now and see if `thud`'s packages no longer require 1.0. 1.1.1 is now the LTS version and has good security-related updates across the board, so I'd be interested in updating to that, but that's not really a critical piece for our image anyway :)

Comment (htot) [reply meta-intel-edison-distro/conf/distro/poky-edison.conf]: Yes thud has 1.1.1a see https://github.com/htot/meta-intel-edison/blob/thud/docs/edison-image-edison.manifest

**Comment by htot** [comment]
Thanks for reviewing.

**Pull request merged**